packages/wasm-utils/src/note/vanchor.rs

```diff
@@ -0,0 +1,76 @@
+use ark_ff::{BigInteger, PrimeField};
+use arkworks_setups::common::{Leaf, VAnchorLeaf};
+use arkworks_setups::r1cs::vanchor::VAnchorR1CSProver;
+use arkworks_setups::utxo::Utxo;
+use arkworks_setups::{Curve as ArkCurve, VAnchorProver};
+use rand::rngs::OsRng;
+
+use crate::note::{JsUtxo, JsUtxoInner};
+use crate::types::{Curve, OpStatusCode, OperationError};
+use crate::VAnchorR1CSProverBn254_30_2_2_2;
+
+pub fn generate_secrets(
+	amount: u128,
+	exponentiation: i8,
+	width: usize,
+	curve: Curve,
+	chain_id: u64,
+	index: Option<u64>,
+	rng: &mut OsRng,
+) -> Result<JsUtxo, OperationError> {
+	let utxo: JsUtxo = match (curve, exponentiation, width) {
+		(Curve::Bls381, 5, 5) => {
```

Comment:
Why are you using Bls381 here? We're only using Bn254 atm. You're even importing the Bn254 tools.